### ISSUE TITLE
[Question] Should URI Health Checks leverage IHttpClientFactory for management?

### ISSUE THREAD

**Issue by joncloud** [Xabaril/AspNetCore.Diagnostics.HealthChecks#30]
Currently the project is creating a new and disposing of `HttpClient` per health check. Microsoft introduced `IHttpClientFactory` to help abstract the lifetime management of `HttpClient` as well as some other benefits in their [original announcement](https://blogs.msdn.microsoft.com/webdev/2018/02/28/asp-net-core-2-1-preview1-introducing-httpclient-factory/). Managing `HttpClient` is also covered in a stack overflow topic: [Do HttpClient and HttpClientHandler have to be disposed?](https://stackoverflow.com/a/15708633)

**Comment by CarlosLanderas** [comment]
@joncloud yes I agree it would be better. especially for HttpClientMessageHandler lifetime delegation. 
We will change it.

**Comment by joncloud** [comment]
If you'd like I can try implementing this.

**Comment by unaizorrilla** [comment]
PR are open!!
